### PR TITLE
Support grpc-web in pure dart

### LINKED ISSUE #216: Enable grpc-web without dart:html
No fundamental reason to limit the grpc-web support to only web-apps as noted by: https://github.com/grpc/grpc-dart/issues/43#issuecomment-510437952 .

### PR DESCRIPTION
**Pull request by robsonmeemo**
Fixes #216

### CODE REVIEW

**Comment by robsonmeemo** [comment]
Is there interest in this PR?

**Comment by sigurdm** [comment]
Yes, this as awesome!

We need to fix the test.

**Comment by robsonmeemo** [comment]
Cool, I wanted to know if there was interest before rewriting the tests. It's done but your Windows tests are failing because of misconfiguration.

FYI, we've been using this branch on a real-world app since March 24th.

**Comment by sigurdm** [comment]
The windows bots seems to work after a reboot.

@nichite are you fine with merging this?

**Comment by robsonmeemo** [comment]
Ping? I'd love to not keep maintaining a fork.

**Comment by galen211** [comment]
This would be useful for a project I'm working on too:  [Related](https://github.com/flutter/flutter/issues/56570)

File: test/client_tests/client_xhr_transport_test.dart [hunk 1]
```diff
@@ -87,33 +68,46 @@ void main() {
...
+          StreamedResponse(Stream.fromIterable([]), 200, headers: metadata));
+    });
+
+    final data = List.filled(10, 0);
+    stream.outgoingMessages.add(data);
 
     stream.incomingMessages.listen((message) {
       expect(message, TypeMatcher<GrpcMetadata>());
       if (message is GrpcMetadata) {
+        print(message.metadata);
```
Comment (sigurdm) [inline test/client_tests/client_xhr_transport_test.dart:110]: Stray print should be removed

File: lib/src/shared/message.dart [hunk 2]
```diff
@@ -69,8 +71,8 @@ StreamTransformer<GrpcMessage, GrpcMessage> grpcDecompressor() =>
         handleData: (GrpcMessage value, EventSink<GrpcMessage> sink) {
       if (value is GrpcData) {
         if (value.isCompressed) {
-          // TODO(dart-lang/grpc-dart#6): Actually handle decompression.
-          sink.add(GrpcData(value.data, isCompressed: false));
+          final decompressedData = GZipDecoder().decodeBytes(value.data);
```
Comment (sigurdm) [inline lib/src/shared/message.dart:74]: Can we take the decompression part out of this PR?

Even though this is awesome I am reluctant to take a dependency on `package:archive`.

File: test/client_tests/client_xhr_transport_test.dart (hunk 1 above)
Comment (robsonmeemo) [reply test/client_tests/client_xhr_transport_test.dart]: done

File: lib/src/shared/message.dart (hunk 2 above)
Comment (robsonmeemo) [reply lib/src/shared/message.dart]: done

**Comment by gustav3d** [comment]
This is awesome !!!

**Comment by gustav3d** [comment]
@robsonmeemo , Do you have any idea what part of your PR that broke server side streaming ?.

**Comment by JohnGalt1717** [comment]
It would be very nice if this package was conditional and would *just work* in flutter so that code was transparent between web and everything else.

**Pull request merged**